addons/controllers/addon_controller.go

```diff
@@ -314,6 +329,19 @@ func (r *AddonReconciler) reconcileNormal(
 	return result, nil
 }
 
+func (r *AddonReconciler) addFinalizer(ctx context.Context, o, deepCopy client.Object) error {
+	controllerutil.AddFinalizer(deepCopy, addontypes.AddonFinalizer)
+	return r.Client.Patch(ctx, deepCopy, client.MergeFrom(o))
+}
+
+func (r *AddonReconciler) removeFinalizer(ctx context.Context, o, deepCopy client.Object) error {
+	if controllerutil.ContainsFinalizer(deepCopy, addontypes.AddonFinalizer) {
```

Comment:
ContainsFinalizer seems redundant since the check is being done before removing as well.

Reply:
will remove

Reply:
done